src/object/cli_obj.c

```diff
@@ -2837,7 +2838,7 @@ dc_obj_fetch(tse_task_t *task, daos_obj_fetch_t *args, uint32_t flags,
 		D_GOTO(out_task, rc);
 	}
 
-	if ((flags & DIOF_EC_RECOV) != 0) {
+	if ((args->extra_flags & DIOF_EC_RECOV) != 0) {
```

Comment:
I do not know how the application can use the "DIOF_EC_RECOV" flag if we export such flag. According to current implementation, this flag is only set by DAOS internal logic. So unless it must be used by application, I will prefer to keep it internally to avoid expose too much detail to up layer application/middleware.

Reply:
OK, it should not be an application known flag, just a pure client internal flag and not a RPC protocol flag.
Seems fine then, thx.